mongomock/collection.py

```diff
@@ -1181,7 +1181,8 @@ def _copy_only_fields(self, doc, fields, container):
                 doc_copy = self._copy_field(doc, container)
         else:
             doc_copy = _project_by_spec(
-                doc, _combine_projection_spec(fields),
+                self._copy_field(doc, container),
```

Comment:
This does the job but in many cases some values will be copied several times. I suggest that you update the _combine_projection_spec instead and add a copy in the places where the are no copies yet (`doc_copy[key] = doc[key]` and `doc_copy[key] = val`)

Reply:
Ah… I think I see what you mean – if fields are excluded (e.g.`find_one(projection={'field': 0})`) or nested fields are used (e.g. `find_one(projection={'field.subfield': 1})`), then `_project_by_spec` would copy the values a second time.

I'll update the pull request accordingly, but I'm wondering a bit about the `container` parameter. As far as I can tell it's never anything but `dict`, so I'm wondering what its purpose is and if it would be OK to just `copy.deepcopy()` the values?